src/pages/detailsServicePage/DetailsServicePage.tsx

```diff
@@ -0,0 +1,164 @@
+import { Grid, Typography, Button, Alert } from '@mui/material'
+import { BarChart } from '../../components/charts/barChart/BarChart'
+import { LineChart } from '../../components/charts/lineChart/LineChart'
+import ArrowBackIcon from '@mui/icons-material/ArrowBack'
+import { InformationBlock } from '../../components/informationBlock/InformationBlock'
+import { ChartsLegend } from '../../components/charts/chartsLegend/ChartsLegend'
+import { useQuery } from '@tanstack/react-query'
+import apiRequests from '../../api/apiRequests'
+import { useTranslation } from 'react-i18next'
+import stores from '../../store/Store'
+import { useNavigate, useParams } from 'react-router-dom'
+import { DetailsServicePageSkeleton } from '../../components/skeleton/DetailsServicePageSkeleton'
+
+const viewInCatalogue = (): void => {
+  console.log('view in catalogue')
+}
+
+export const DetailsServicePage: React.FC = () => {
+  const { t } = useTranslation(['general', 'detailsPage'])
+
+  const navigate = useNavigate()
+
+  const params = useParams()
+
+  const [updateSnackbar] = stores.useSnackbarStore((state) => [state.updateSnackbar])
+
+  // elements for the legend component
+  const legendElements = [
+    { label: 'E-service online', color: '#17324D' },
+    { label: t('monitoringSuspended', { ns: 'detailsPage' }), color: '#A2ADB8' },
+    { label: 'E-service offline', color: '#FE6666' },
+  ]
+
+  const eserviceRecordId = params.id ?? ''
+
+  async function fetchServiceMainData(value: string): Promise<any> {
```

Comment:
Evitare l'utilizzo del type `any` se non ci sono particolari ragioni.

Ho visto che i types dei dati che arrivano da queste chiamate sono dichiarati nei componenti in cui vengono passati.
Perché non rendere questi modelli dati importabili ed utilizzabili globalmente?

Nel pratico, il mio consiglio è il seguente: 

Il modello dati `MainData` è dichiarato nel componente `InformationBlock`. 
Spostare la dichiarazione di questo tipo in un eventuale file `src/types/e-service.types.ts`(o `src/types.ts` se non ne prevedete molti).

``` typescript
export interface MainData {
  eserviceName: string
  producerName: string
  versionNumber: string
}
```

Rimpiazzare poi questo any con quel type.

```suggestion
  async function fetchServiceMainData(value: string): Promise<MainData> {
```

useQuery poi under-the-hood farà automaticamente l'inference, quindi i dati che ritornano dall'hook avranno il type specificato, in questo caso quindi la costante `mainData` sarà di tipo `MainData`.